libgalois/include/katana/PropertyManager.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "katana/Manager.h"
+#include "katana/MemoryManager.h"
+
+/// Managers tend to the memory consumption for specific, large resources, e.g.,
+/// properties or views.  They interact with the central MemoryManager (MM singleton)
+/// to coordinate memory use.
+
+class PropertyManager : public Manager {
+public:
+  /// Returns the coarse category of memory use
+  ///   e.g., property for the property manager
+  virtual const std::string& MemoryCategory() { return "property"; }
+
+  /// Free standby memory, attempting to free \p goal bytes.
+  /// Returns the number of bytes freed, which can only be less than goal if the
+  /// manager's standby total is less than goal.
+  virtual count_t FreeStandbyMemory(count_t goal);
+
+  // void Insert(
+  //     const katana::Uri& cache_key, const std::shared_ptr<arrow::table>& props);
+
+  void AddProperty(const katana::Uri& property_path) {
+    auto property = cache_->Get(property_path);
+    if (property.has_value()) {
+      auto bytes = katana::ApproxTableMemUse(property.value());
+      MM().StandbyToActive(this, bytes);
```

Comment:
Didn't understand how this is compiling. Shouldn't this be `MemoryManager::MM()...`?

Reply:
Yes it should.  This code isn't active yet.